save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/UsersDetailsController.kt

```diff
@@ -0,0 +1,54 @@
+package org.cqfn.save.backend.controllers
+
+import org.cqfn.save.backend.repository.UserRepository
+import org.cqfn.save.domain.ImageInfo
+import org.cqfn.save.info.UserInfo
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.*
+import reactor.core.publisher.Mono
+
+/**
+ * Controller that handles operation with users
+ */
+@RestController
+@RequestMapping("/api/users")
+class UsersDetailsController(
+    private val userRepository: UserRepository,
+) {
+    /**
+     * @param userName username
+     * @return [ImageInfo] about user's avatar
+     */
+    @GetMapping("/{userName}/avatar")
+    @PreAuthorize("permitAll()")
+    fun avatar(@PathVariable userName: String): Mono<ImageInfo> = Mono.fromCallable {
+        userRepository.findByName(userName).get().avatar
+            .let { ImageInfo(it) }
+    }
+
+    /**
+     * @param userName username
+     * @return [UserInfo] info about user's
+     */
+    @GetMapping("/{userName}")
+    @PreAuthorize("permitAll()")
+    fun findByName(@PathVariable userName: String): Mono<UserInfo> = Mono.fromCallable {
+        userRepository.findByName(userName).get().toUserInfo()
+    }
+
+    /**
+     * @param newUserInfo
+     */
+    @PostMapping("/save")
+    fun saveUser(@RequestBody newUserInfo: UserInfo) {
```

Comment:
Any permission checks here?

Reply:
This method is called only by the user for himself. Are there any other permission checks needed?

Reply:
Does anything stop me from doing `curl -d '{"name": "Cheshiriks"}' saveourtool.com/api/users/save`?

Reply:
Added annotation

Reply:
Again, I can be authenticated as myself and still change your user data. You should check if the user from `Authentication` object is the same that the user in `UserInfo`.

Reply:
Ok. Fixed